ush/forecast_predet.sh

```diff
@@ -52,9 +52,13 @@ common_predet(){
   previous_cycle=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} - ${assim_freq} hours" +%Y%m%d%H)
   next_cycle=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} + ${assim_freq} hours" +%Y%m%d%H)
   current_cycle_begin=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} - ${half_window} hours" +%Y%m%d%H)
-  current_cycle_end=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} + ${half_window} hours" +%Y%m%d%H)
-  next_cycle_begin=$(date --utc -d "${next_cycle:0:8} ${next_cycle:8:2} - ${half_window} hours" +%Y%m%d%H)
-  next_cycle_end=$(date --utc -d "${next_cycle:0:8} ${next_cycle:8:2} + ${half_window} hours" +%Y%m%d%H)
+  if [[ "${DOIAU:-}" == "YES" ]]; then
```

Comment:
@aerorahul my test is still going, but this seems to be working as expected with this fix.  If current_cycle_begin and next_cycle_begin should not change definitions, we could add a variable here perhaps startdate_current_cycle startdate_next_cycle that could be defined and updated as current_cycle_begin/next_cycle_begin are being defined here?

Reply:
I am working on a patch that will:
1. have the ability to control if we want to save last time restarts
2. save restarts at the appropriate time of next cycle when doing IAU, but the experiments start  with a cold start

Reply:
What are the issues with the solution (or alternative) shared here?

Reply:
I don't believe this is correct.
Regardless of DOIAU, current_cycle_begin is the time of the beginning of the assimilation window. If it makes it more clear to call this current_cycle_window_begin, then we can rename w/ `window` in the word.
current_cycle and next_cycle are CDATE and CDATE+6h.  Those are fixed.

Reply:
So what I understand is that you would like for the variables  current_cycle_begin to not be changed.   In this case, I can add the new variables: startdate_current_cycle startdate_next_cycle or a name you would prefer that would allow for current_cycle_begin/next_cycle_begin to remain unchanged.    Is startdate_current_cycle startdate_next_cycle  acceptable names?

Reply:
I've made an update that hopefully addresses your concerns.  Please let me know if there are other outstanding concerns I can address.

Reply:
Yes. Thank you.   
`sdate_current_cycle` is the `model_start_date_current_cycle` and `sdate_next_cycle` is the `model_start_date_next_cycle`. And this is captured here rather than the many places where `DOIAU` is relevant.

Reply:
I will leave it to you; a descriptive name is useful.

Reply:
I updated the name so it's more descriptive.

Reply:
Thanks for cleaning this up.  It looks great.